### ISSUE TITLE
Problem with 3D calibration

### ISSUE THREAD

**Issue by aaristov** [jungmannlab/picasso#20]
I try to calibrate astigmatism from the stack https://www.dropbox.com/s/q565b0bcjyz1f12/cal-ast-5nm_1-crop.tif?dl=0

And polynomial fit looks weird:
https://www.dropbox.com/s/emo8my51tund3ud/3d_cal.png?dl=0

I set the following parameters:
Box size 16
min Net Grad 810
EM gain 1 (data in photon counting mode)
Baseline 0
pixel size 110
Method: LQ
Step size: 5 nm

I also copy the terminal output:

> C:\Users\Andrey\Anaconda3\envs\picasso\lib\site-packages\numpy\core\_methods.py:
> 59: RuntimeWarning: Mean of empty slice.
>  warnings.warn("Mean of empty slice.", RuntimeWarning)
> C:\Users\Andrey\Anaconda3\envs\picasso\lib\site-packages\numpy\core\_methods.py:
> 70: RuntimeWarning: invalid value encountered in true_divide
>  ret = ret.dtype.type(ret / rcount)

> Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.
> C:\Users\Andrey\Anaconda3\envs\picasso\lib\site-packages\numpy\lib\polynomial.py
> :595: RankWarning: Polyfit may be poorly conditioned
>   warnings.warn(msg, RankWarning)

> Intel MKL ERROR: Parameter 6 was incorrect on entry to DGELSD.
> C:\Users\Andrey\Anaconda3\envs\picasso\lib\site-packages\numpy\lib\polynomial.py
> :595: RankWarning: Polyfit may be poorly conditioned
>  warnings.warn(msg, RankWarning)
> C:\Users\Andrey\Anaconda3\envs\picasso\lib\site-packages\numpy\core\_methods.py:
>59: RuntimeWarning: Mean of empty slice.
>  warnings.warn("Mean of empty slice.", RuntimeWarning)
>C:\Users\Andrey\Anaconda3\envs\picasso\lib\site-packages\numpy\core\_methods.py:
>70: RuntimeWarning: invalid value encountered in double_scalars
>  ret = ret.dtype.type(ret / rcount)
>QCoreApplication::exec: The event loop is already running

**Comment by straussmaximilian** [comment]
Hi, I could reproduce the problem. Apparently, the fit is not converging, and this is causing problems.
I noticed that your PSFs are quite large, so I did a sum-binning in ImageJ and was able to get a better polynomial fit:

![screen shot 2018-06-18 at 5 20 18 pm](https://user-images.githubusercontent.com/16521288/41545763-3bcf3d34-731c-11e8-85c5-894b61b97de4.png)

I would recommend that you activate binning on your camera. Additionally, you could increase the z-range (from, e.g. -500 to +500). Note that Picasso always assumes that you are going from minus to plus.

**Comment by aaristov** [comment]
There is localization filter in the code which removes some localizations with low certainty, but z_range vector stays unaffected. That's why polyfit doesn't work. Pull request sent.

**Comment by straussmaximilian** [comment]
Hi,
I checked this again and noticed the nans. I guess we never experienced that as we usually have more spots and higher SNR. I now made an additional check that interpolates the nans, and the fit worked smoothly. Feel free to check out #995 in the dev branch.

<img width="980" alt="screen shot 2018-06-18 at 10 20 20 pm" src="https://user-images.githubusercontent.com/16521288/41560429-5092407e-7347-11e8-8f4c-51247a0ab6ce.png">
